src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -252,6 +237,52 @@ internal void SetMaxMetricPointsPerMetricStream(int maxMetricPointsPerMetricStre
         }
     }
 
+    private static void DeactivateMetric(Metric metric)
+    {
+        metric.InstrumentDisposed = true;
+
+        var metricStreamIdentity = metric.InstrumentIdentity;
```

Comment:
You could remove this and simply use `metric.Name` and `metric.MeterName`.

Reply:
I think it is a bit slower but I made the change since it isn't perf critical area.